### DISCUSSION TITLE
Inconsistent Flight with Optitrack

### DISCUSSION THREAD

**Discussion by sesem738** [bitcraze/discussions#1088] [Q&A]
Hello,

I’ve been experimenting with the Crazyflie drone in conjunction with Optitrack, but unfortunately, the results have been quite inconsistent. The majority of the flights have been unsuccessful, with only occasional successful ones. This inconsistency is also observed when running the mocap_hl_commander example. I’m beginning to suspect that the issue might lie with the Optitrack system, because I am able to fly the autonomousSequence example with the FlowDeck consistent. But I’m unsure of how to proceed with debugging it.

**Comment by knmcguire** [comment]
Hi! So there are many reasons why optitrack is not working for you:
- The cameras are too far away. How big is your tracking area?
- The markers are too small, dirty or not visibile
- There is a lot of daylight in your flight area. Mocap works with infrared light so that will be an issue
Perhaps optitrack also has instructions on how to improve the quality. We don't have optitrack ourselves so we don't have a lot of tips, but you can probe the crazyswarm2 discussions as there are more mocap flyers there with the crazyflie.

You can also take look directly at the libmotioncapture and use some examples there: https://github.com/IMRCLab/libmotioncapture which is the base of the example of the[ mocap_hl_commander example](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py). Note that the same lab also have a library that does single marker tracking as well, which might be more robust in some situations: https://github.com/IMRCLab/librigidbodytracker
